corehq/apps/receiverwrapper/views.py

```diff
@@ -209,8 +209,10 @@ def _record_metrics(tags, submission_type, response, timer=None, xform=None):
 @waf_allow('XSS_BODY')
 @csrf_exempt
 @api_auth
+@require_permission(Permissions.edit_data)
+@require_permission(Permissions.access_api)
 def post_api(request, domain, app_id=None):
-    return post(request, domain, app_id)
+    return secure_post(request, domain, app_id)
```

Comment:
side note: I don't really like that these views all call each other and re-apply a bunch of the decorators (including sometimes auth).

Reply:
yeah, agreed - I especially don't like that form submissions handles auth its own way, differently than everything else.

Reply:
oh wait - since this _does_ define a definite auth decorator (`@api_auth`), I think it can just call `_process_form` - there's no need for `secure_post`.  Not sure why I didn't think of that before.  I'll try it out and make a followup PR.

Reply:
I was going to suggest that but it would also require copying some of the other decorators like `require_POST`, `check_domain_migration`, `set_request_duration_reporting_threshold`. But that is probably preferable.